### PR TITLE
Mimic publish date UI

### LINKED ISSUE #4: Date picker or mimic publish date UI
Using a simple text field makes it easy to enter an invalid date.
We could copy the publish date UI fields or use a datepicker

### PR DESCRIPTION
**Pull request by kucrut**
screen shot 2016-07-25 at 2 33 30 pm

Fixes #4.

### CODE REVIEW

**Comment by kucrut** [comment]
@peterwilsoncc Could you be so kind and review this PR? Thanks!

**Comment by peterwilsoncc** [comment]
In the process of reviewing.

From a UI perspective, it would be nice if the JS updated the display when the user hit ok, cancel or clear. The updated element would be `#unpublish-timestamp > strong`. Hopefully this is adaptable from the existing UI for the publish time.

File: inc/templates/unpublish-ui.tpl.php [hunk 1]
```diff
@@ -1,21 +1,55 @@
...
-			$(this).next().show();
-		});
+	<fieldset id="unpublish-timestampdiv" class="hide-if-js">
+		<legend class="screen-reader-text"><?php esc_html_e( 'Unpublish Date and time', 'unpublish' ); ?></legend>
+		<div class="unpublish-timestamp-wrap">
+			<label>
+				<span class="screen-reader-text"><?php esc_html_e( 'Month', 'unpublish' ); ?></span>
+				<select id="unpublish-mm" name="unpublish-mm">
+					<option value=""><?php esc_html_e( '&mdash;', 'unpublish' ); ?></option>
+						<?php foreach ( $month_names as $month ) : ?>
```
Comment (peterwilsoncc) [inline inc/templates/unpublish-ui.tpl.php]: CS: there's an extra tab here.

File: js/unpublish.js [hunk 2]
```diff
@@ -0,0 +1,46 @@
...
+		$fieldset.find( ':input' ).val( '' );
+		$stamp.html( '&mdash;' );
+	});
+
+	$( '.save-unpublish-timestamp' ).on( 'click', function( e ) {
+		e.preventDefault();
+		$editLink.show();
+		hideFieldset();
+
+		// TODO: Should we validate the date parts here?
```
Comment (peterwilsoncc) [inline js/unpublish.js]: I would to match the publish date field as closely as possible. A lot of the code will be adaptable from `/wp-admin/js/post.js` in core.

File: inc/templates/unpublish-ui.tpl.php [hunk 3]
```diff
@@ -1,21 +1,55 @@
...
+				<a href="#edit-unpublish-timestamp" class="cancel-unpublish-timestamp hide-if-no-js button-cancel"><?php esc_html_e( 'Cancel', 'unpublish' ); ?></a>
+				<a href="#edit-unpublish-timestamp" class="clear-unpublish-timestamp hide-if-no-js button-cancel" title="<?php esc_attr_e( 'Clear unpublish date', 'unpublish' ); ?>"><?php esc_html_e( 'Clear', 'unpublish' ); ?></a>
+			</p>
+		</div>
+	</fieldset>
 
-	});
-</script>
\ No newline at end of file
+	<input type="hidden" name="unpublish-nonce" value="<?php echo esc_attr( wp_create_nonce( 'unpublish' ) ); ?>" />
```
Comment (peterwilsoncc) [inline inc/templates/unpublish-ui.tpl.php]: `wp_nonce_field()`

File: unpublish.php [hunk 4]
```diff
@@ -74,49 +74,126 @@ public function action_load_customizations() {
...
+	public function enqueue_scripts_styles() {
+		wp_enqueue_style( 'unpublish', plugins_url( 'css/unpublish.css', __FILE__ ), array(), '0.1-alpha' );
+		wp_enqueue_script( 'unpublish', plugins_url( 'js/unpublish.js', __FILE__ ), array( 'jquery' ), '0.1-alpha', true );
 	}
 
 	/**
 	 * Save the unpublish time for a given post
 	 */
 	public function action_save_unpublish_timestamp( $post_id ) {
+		if ( ! isset( $_POST['unpublish-nonce'] ) || ! wp_verify_nonce( $_POST['unpublish-nonce'], 'unpublish' ) ) {
```
Comment (peterwilsoncc) [inline unpublish.php]: `check_admin_referer()` - it handles the nonce and referrer checking.

**Comment by peterwilsoncc** [comment]
A few minor notes re style. Mainly the JS needs extending to more closely emulate  the behaviour of the publish field's behaviour.

**Comment by kucrut** [comment]
@peterwilsoncc I've made some fixes and added date validation.

**Comment by peterwilsoncc** [comment]
I've started getting a bug where I can't hit the content type when I enable unpublishing. I'll take a look to see if anything jumps out but I'm well confused now.

This happens with both network enabling and enabling on the site only.

are-you-sure-error

**Comment by peterwilsoncc** [comment]
@kucrut Sorry, gave you a bum steer on `check_admin_referer`. It dies rather than returns and WP runs the filter when you hit the new post screen. I've reverted it.

File: unpublish.php (hunk 4 above)
Comment (peterwilsoncc) [inline unpublish.php]: This statement is incorrect, leave as is.

(resolved)

**Pull request merged**